lib/auth/person.ex

```diff
@@ -272,40 +277,63 @@ defmodule Auth.Person do
     end
   end
 
-  @doc """
-  `list_people/0` lists all people in the system.
-  Used for displaying the table of authenticated people.
-  """
-  def list_people do
-    Repo.all(from(pr in __MODULE__, preload: [:roles, :statuses]))
-    # keeping this query commented here for now in case I decide to use it
-    # instead of having to call PeopleView.status_string/2
-    # query = """
-    # SELECT DISTINCT ON (s.status_id, s.person_id) s.id, s.message_id,
-    # s.updated_at, s.template, st.text as status, s.person_id
-    # FROM sent s
-    # JOIN status as st on s.status_id = st.id
-    # WHERE s.message_id IS NOT NULL
-    # """
-    # {:ok, result} = Repo.query(query)
-
-    # # create List of Maps from the result.rows:
-    # Enum.map(result.rows, fn([id, mid, iat, t, s, pid]) ->
-    #   # e = Fields.AES.decrypt(e)
-    #   # e = case e !== :error and e =~ "@" do
-    #   #   true -> e |> String.split("@") |> List.first
-    #   #   false -> e
-    #   # end
-    #   %{
-    #     id: id,
-    #     message_id: mid,
-    #     updated_at: NaiveDateTime.truncate(iat, :second),
-    #     template: t,
-    #     status: s,
-    #     person_id: pid,
-    #     email: ""
-    #   }
-    # end)
-    # |> Enum.sort(&(&1.id > &2.id))
+  # @doc """
+  # `get_list_of_people/1` retrieves the list of people
+  # that a given logged in person can see for all the Apps they have.
+  # Used for displaying the table of authenticated people.
+  # """
+  def get_list_of_people(conn) do
+    # IO.inspect(conn.assigns.person)
+    apps = Enum.map(Auth.App.list_apps(conn), fn a -> a.id end)
+    app_ids = if length(apps) > 0, do: Enum.join(apps, ","), else: "0"
+
+    query = """
+    SELECT l.app_id, l.person_id, p.status,
+    st.text as status, p."givenName", p.picture,
+    l.inserted_at, p.email, l.auth_provider, r.name
+    FROM (
+      SELECT DISTINCT ON (person_id) *
+      FROM logs
+      ORDER BY person_id, inserted_at DESC
+    ) l
+    JOIN people as p on l.person_id = p.id
+    LEFT JOIN status as st on p.status = st.id
+    LEFT JOIN people_roles as pr on p.id = pr.person_id
+    LEFT JOIN roles as r on pr.role_id = r.id
+    WHERE l.app_id in (#{app_ids})
+    ORDER BY l.inserted_at DESC
+    NULLS LAST
+    """
+
+    {:ok, result} = Repo.query(query)
```

Comment:
Very happy to add error handling or switch to `Repo.query!/3` (_I wrote this at 1am after a long day..._ 😉 )
Worth noting that when the result set is empty it just returns empty array so the rest of the code still works.
(_it doesn't throw a nasty error even if there is no data_)

Reply:
Yeah this would only change the failure point of a database error from this pattern match to Ecto, which might result in a clearer error if things do go wrong

Reply:
Updated to use `Repo.query!` ✅